site/content/blog/2025-02-17-portswigger-lab-walkthrough-broken-brute-force-protection-ip-block/index.md

```diff
@@ -0,0 +1,793 @@
+
+Our code sample above imports the `Thread` class using `Java.type('java.lang.Thread')`. We’ve used it to implement the one-minute delay between requests. For more details on available JavaScript features and the Java APIs accessible in your script, see [GraalJS Compatibility](https://www.graalvm.org/latest/reference-manual/js/JavaScriptCompatibility/) and [Java Interoperability](https://www.graalvm.org/jdk21/reference-manual/js/JavaInteroperability/).
+
+If you’ve modified the script as stated, save it and ensure it is enabled. Aside from right-clicking the script's name in the *Scripts* tab, you can also enable it with the green check icon in the *Scripts Console* tab.
+
+#### Step 3: Add the Payload
+
+Next, let’s open the Fuzzer dialog. Right-click the captured POST request from the *Sites* tab and select *Attack* > *Fuzz*. In the *Fuzzer Locations* tab, we’ll select the password value as the position we want to fuzz. Highlight it to activate the *Add* button:
+
+&nbsp;
+![fuzz-password.png](images/fuzz-password.png)
+&nbsp;
+
+In the *Add Payload* dialog, keep the default “Strings” payload generator type. Then, copy the [password wordlist](https://portswigger.net/web-security/authentication/auth-lab-passwords) from the lab page and paste it into the *Contents* field:
+
+&nbsp;
+![add-password-payload.png](images/add-password-payload.png)
+&nbsp;
+
+Save the payload and return to the main Fuzzer dialog.
+
+#### Step 4: Add the Message Processor Script
+
+Finally, switch to the *Message Processors* tab. This is where we will add our IP block handler script. Click *Add* to bring up the *Add Message Processor* dialog and select your script:
+
+&nbsp;
+![move-message-processor-script-top.png](images/move-message-processor-script-top.png)
+&nbsp;
+
+You can optionally move the script to the top of the processor list. Then, run the Fuzzer.
+
+<a id="have-username-got-password"></a>
+## Have Username, Got Password
+
+Once the Fuzzer is done, we can inspect the results. We’re looking for indications of a successful login. So, we’ll sort the results in descending order using the status code column (simply “*Code”* in the UI).
+
+### Method 1: Interleave Wordlists With Valid Credentials
+
+Our sorted results from our bypass with valid credentials show multiple `302 Found` results. This is because of the logins to our own account. So, we want to search the *Payloads* column for a payload that includes “carlos”:
+
+&nbsp;
+![302_found_carlos.png](images/302_found_carlos.png)
+&nbsp;
+
+Voila! Carlos’ password is “soccer”. Log in with the password to solve the lab.
+
+### Method 2: Wait Out the IP Block
+
+For the second method, we also want to sort the results in descending order by status code. At the top of our sorted results is a `302 Found`:
+
+&nbsp;
+![password-found.png](images/password-found.png)
+&nbsp;
+
+The password is “biteme”, as we can see either through the *Payloads* column or in the request body:
```

Comment:
Which is it? The text and screenshots need to be updated with the correct finding.

Reply:
Ah, I wasn't sure how to handle this section. Since I solved the lab twice with two different methods, there are two different passwords and slightly different ways to locate them (since the first password is mixed up in other 302 results, unlike the second). The images show this difference and the text is suppose to complement them. Do you think I should approach this differently?

Reply:
Do both passwords actually solve the lab?

Reply:
Yeah, both solutions are valid. But the actual password is not necessarily going to be the same for each reader. I've solved some labs multiple times. Same method, different lab instances, different passwords. (I think each lab instance is available for a couple hours max, so if you're done by then, you have to start up a new one). Not sure about the technical details of how the labs are set up.

Reply:
Ok so just make sure to add a callout note/warning that clarifies this.

Reply:
Makes sense, thanks.